### DISCUSSION TITLE
How DAKOTA (MOGA algorithm) works?

### DISCUSSION THREAD

**Discussion by rafeemahbub** [snl-dakota/dakota#46] [Q&A]
I am a graduate student at Marquette University. I have some questions regarding how DAKOTA works, specifically the MOGA algorithm. It would be very helpful if anyone could help me with the answers to the following questions. I would appreciate your help. 

We have introduced an optimization algorithm, MOGA: Multi-Objective Genetic Algorithm in the DAKOTA platform to access the best simulation parameters of a material. In each new generation, DAKOTA selects some parameters value and runs simulation software to compare with the given objective functions.

I have some questions regarding how the DAKOTA is performing the optimizations. The questions are given below: 
 
1.	What does crossover mean?  
2.	Does it simply swap variables within the DNA?  How many does it swap? 
3.	How do we control this process?  How do we control how many children are born with crossover?  How do we control how many parents are used to generate one child?
4.	How does Mutation work?  How do we control how many children are born with mutations?  
5.	How do we control the variations of the mutation?  Is it perturbed (mutated) with a Gaussian function?  How do we control the number of variables within a child that is mutated?
6.	Is Dakota making children that only swap or only have a mutation?  Or is it mixing swapping and mutations within a single child?

**Comment by jadamstephens** [comment]
Hi,

The description of the MOGA algorithm in Dakota's [keyword reference](https://snl-dakota.github.io/docs/6.18.0/users/usingdakota/reference/method-moga.html) explains reasonably thoroughly how the overall method works and how the various controls affect the way operators such as crossover and mutation work.

(These pages need some formatting fixes, and you may be better off looking at the Reference Manual for an older version of Dakota, such as this one for the [6.16 release](https://www.sandia.gov/app/uploads/sites/241/2023/03/html-ref-6.16.0.tar.gz). The documentation for MOGA hasn't changed since then.)

A common mistake novice Dakota uses make when using MOGA is not realizing that MOGA is a **multiobjective** optimizer and is meant for Pareto optimization. If you have only a single objective function, you should instead use SOGA.
